File: leaoai_python_refatorado_descompactado/app/models/Factory/projectimage.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.core.database import Base
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .projectblock import ProjectBlock

class ProjectImage(Base):
    __tablename__ = "ProjectImages"

    Id = Column(Integer, primary_key=True, index=True)
    ProjectId = Column(Integer, nullable=True, ForeignKey('projects.Id'))
    ProjectBlockId = Column(Integer, nullable=True, ForeignKey('projectblocks.Id'))
    UrlSource = Column(String)
    DataCadastro = Column(DateTime)
    description = Column(String, nullable=True)
    enableOnReport = Column(Boolean)

    # Define relationships here
    ProjectBlock = relationship('ProjectBlock')

File: Framework.Data/Models/Factory/ProjectImage.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Framework.Data.Models.Factory
{
	// PROJECT IMAGE
	public class ProjectImage
	{
		[Key]
		public int Id { get; set; }

		[ForeignKey("Project")]
		public int ProjectId { get; set; }
		public virtual Project Project { get; set; }

		[ForeignKey("ProjectBlock")]
		public int ProjectBlockId { get; set; }
		public virtual ProjectBlock ProjectBlock { get; set; }

		[Required]
		public string UrlSource { get; set; }
		public DateTime DataCadastro { get; set; } = DateTime.UtcNow;

		public bool enableOnReport { get; set; } = false;
		public string description { get; set; }
	}
}
